gui/integrity-dialog.cpp

```diff
@@ -146,6 +146,37 @@ IntegrityDialog::IntegrityDialog(Common::String endpoint, Common::String domain)
 IntegrityDialog::~IntegrityDialog() {
 }
 
+
+bool IntegrityDialog::progressUpdateCallback(int bytesProcessed) {
+	if (g_checksum_state->dialog->_close)
+		return false;
+	
+	g_checksum_state->calculatedSize += bytesProcessed;
+
+	if (g_system->getMillis() > g_checksum_state->lastUpdate + 500) {
+		g_checksum_state->lastUpdate = g_system->getMillis(); 
+		g_checksum_state->dialog->sendCommand(kDownloadProgressCmd, 0);
+	}
+
+	Common::Event event;
+	g_checksum_state->dialog->pollEvent(event);
```

Comment:
what is the reason of moving it out to a separate method? I see it is used only in this place